Packs/illuminate/Integrations/Analyst1/Analyst1.py

```diff
@@ -29,10 +26,426 @@
+
+    Returns:
+        The nested value if found, otherwise None
+    """
+    top_level = data.get(key)
+    if isinstance(top_level, dict):
+        return top_level.get(nested_key)
+    return None
+
+
+def get_analyst1_tags_for_batch_result(results_for_indicator: list[dict]) -> list[str]:
+    """
+    Determines which Analyst1 tags should be applied to an indicator based on batch check results.
+
+    An indicator can have multiple entity types (e.g., both "Asset" and "In Home Range"),
+    so this function collects all applicable tags from all results for the same indicator.
+
+    Args:
+        results_for_indicator: List of batch check result objects for a single indicator
+
+    Returns:
+        List of tag strings that should be applied (e.g., ["Analyst1: Asset", "Analyst1: Indicator"])
+    """
+    tags = set()
+
+    for result in results_for_indicator:
+        entity = result.get("entity", {})
+        entity_key = entity.get("key") if isinstance(entity, dict) else None
+        if entity_key and entity_key in ENTITY_TYPE_TAGS:
+            tags.add(ENTITY_TYPE_TAGS[entity_key])
+
+    return list(tags)
+
+
+def get_xsoar_indicator_type_from_batch_result(result: dict) -> str:
+    """
+    Determines the XSOAR indicator type from a batch check result.
+
+    Args:
+        result: A single batch check result object
+
+    Returns:
+        XSOAR indicator type string (e.g., "email", "ip", "domain")
+        Returns "unknown" if the type cannot be determined
+    """
+    analyst1_type = get_nested_value(result, "type", "key")
+    if analyst1_type and analyst1_type in ANALYST1_TO_XSOAR_TYPE:
+        return ANALYST1_TO_XSOAR_TYPE[analyst1_type]
+    return "unknown"
+
+
+def calculate_batch_check_verdict(entity_key: str | None, risk_score: str | None, benign_value: bool | None, params: dict) -> int:
+    """
+    Calculates XSOAR verdict score for batchCheck results based on entity.key, risk score, and benign flag.
+
+    Priority logic:
+    1. If benign=True, always return Benign (1)
+    2. If entity.key is ASSET/IN_SYSTEM_RANGE/IN_HOME_RANGE/IN_PRIVATE_RANGE/IGNORED_INDICATOR/IGNORED_ASSET → Benign (1)
+    3. If entity.key is INDICATOR and risk score exists, map it using configuration
+    4. If risk score is null, return Unknown (0)
+
+    Args:
+        entity_key: Analyst1 entity.key value from batchCheck response
+        risk_score: Analyst1 risk score value (Lowest, Low, Moderate, High, Critical, Unknown)
+        benign_value: Analyst1 benign flag value
+        params: Integration configuration parameters
+
+    Returns:
+        XSOAR verdict score: 0=Unknown, 1=Benign, 2=Suspicious, 3=Malicious
+    """
+    # Priority 1: benign=True always results in Benign verdict
+    if benign_value is True:
+        return 1  # Benign
+
+    # Priority 2: Check entity.key for benign categories
+    benign_entity_keys = ["ASSET", "IN_SYSTEM_RANGE", "IN_HOME_RANGE", "IN_PRIVATE_RANGE", "IGNORED_INDICATOR", "IGNORED_ASSET"]
```

Comment:
Consider defining `benign_entity_keys` as a global constant to avoid duplication and ensure consistency. This list is also defined in the `has_benign_entity_type` function on line 257.